pkg/decoders/escaped_unicode.go

```diff
@@ -23,6 +23,33 @@ var (
 
 	// Common escape sequence used in programming languages.
 	escapePat = regexp.MustCompile(`(?i:\\{1,2}u)([a-fA-F0-9]{4})`)
+
+	// Additional Unicode escape formats from dencode.com
+
+	// \u{X} format - Rust, Swift, some JS, etc. (variable length hex in braces)
+	braceEscapePat = regexp.MustCompile(`\\u\{([a-fA-F0-9]{1,6})\}`)
+
+	// \U00XXXXXX format - Python, etc. (8-digit format for non-BMP characters)
+	longEscapePat = regexp.MustCompile(`\\U([a-fA-F0-9]{8})`)
+
+	// \x{X} format - Perl (variable length hex in braces)
+	perlEscapePat = regexp.MustCompile(`\\x\{([a-fA-F0-9]{1,6})\}`)
+
+	// \X format - CSS (hex without padding). Go's regexp (RE2) has no look-ahead, so we
+	// include the delimiter (whitespace, another backslash, or end-of-string) in the
+	// match using a non-capturing group. The delimiter is later re-inserted by the
+	// decoder when necessary.
+	cssEscapePat = regexp.MustCompile(`\\([a-fA-F0-9]{1,6})(?:\s|\\|$)`)
```

Comment:
If these are specific to Perl and CSS I think we could probably omit them for better performance